src/hotspot/cpu/ppc/c1_LIRAssembler_ppc.cpp

```diff
@@ -540,46 +540,26 @@ void LIR_Assembler::emit_opConvert(LIR_OpConvert* op) {
     }
     case Bytecodes::_i2d:
     case Bytecodes::_l2d: {
-      bool src_in_memory = !VM_Version::has_mtfprd();
       FloatRegister rdst = dst->as_double_reg();
-      FloatRegister rsrc;
-      if (src_in_memory) {
-        rsrc = src->as_double_reg(); // via mem
+      // move src to dst register
+      if (code == Bytecodes::_i2d) {
+        __ mtfprwa(rdst, src->as_register());
       } else {
-        // move src to dst register
-        if (code == Bytecodes::_i2d) {
-          __ mtfprwa(rdst, src->as_register());
-        } else {
-          __ mtfprd(rdst, src->as_register_lo());
-        }
-        rsrc = rdst;
+        __ mtfprd(rdst, src->as_register_lo());
       }
-      __ fcfid(rdst, rsrc);
+      __ fcfid(rdst, rdst);
       break;
```

Comment:
Would it make sense to split this code into the two separate switch cases `_i2d` and `l2d`, now that the code has fewer lines?